### PR TITLE
Python3 build

### PR DESCRIPTION
**Pull request by paskino**
Updates scripts and packages to build with SIRF with python3.

### CODE REVIEW

**Comment by paskino** [comment]
This also means that some packages in the SuperBuild pick the wrong python version and I'm not entirely sure which one.

**Comment by KrisThielemans** [comment]
I think this will be premature to merge for 2.2. We can of course always release another VM afterwards.

Possibly this could be resolved by tackling https://github.com/SyneRBI/SIRF-SuperBuild/issues/48.

**Comment by rijobro** [comment]
As discussed previously, we could potentially switch to 20.04 meaning that it would use python3 by default. Might fix some problems without any effort on our part.

Would require fixing https://github.com/SyneRBI/SIRF-SuperBuild/issues/413.

**Comment by KrisThielemans** [comment]
> without any effort on our part

seems somewhat wishful thinking, given Alex's struggles :wink:

In any case, finding correct Python versions will remain a bane. Maybe not on a clean Ubuntu 20.04 VM, but as soon as conda gets involved, it'll be a load of fun to be sure we don't mix conda and system python.

File: scripts/UPDATE.sh [hunk 1]
```diff
@@ -194,8 +195,10 @@ SuperBuild(){
         -DBUILD_siemens_to_ismrmrd=On \
         -DUSE_ITK=ON \
         -DDEVEL_BUILD=OFF\
+        -DNIFTYREG_USE_CUDA=OFF\
+        -DPYVER:STRING=3\
         -DBUILD_CIL_LITE=ON\
-        -DNIFTYREG_USE_CUDA=OFF
+	-DCYTHON_EXECUTABLE=cython3
```
Comment (KrisThielemans) [inline scripts/UPDATE.sh:201]:
```suggestion
	-DCYTHON_EXECUTABLE=/usr/bin/cython3
	-DPYTHON_EXECUTABLE=/usr/bin/python3
```

File: scripts/UPDATE.sh [hunk 2]
```diff
@@ -194,8 +195,10 @@ SuperBuild(){
         -DBUILD_siemens_to_ismrmrd=On \
         -DUSE_ITK=ON \
         -DDEVEL_BUILD=OFF\
+        -DNIFTYREG_USE_CUDA=OFF\
+        -DPYVER:STRING=3\
```
Comment (KrisThielemans) [inline scripts/UPDATE.sh:199]:
```suggestion
```

**Review by KrisThielemans — CHANGES_REQUESTED** [review]

File: scripts/UPDATE.sh (hunk 2 above)
Comment (paskino) [reply scripts/UPDATE.sh:199]: Why do you suggest to remove `-DPYVER=3`? Or you suggest to remove forcing it into the cache?

Comment (KrisThielemans) [reply scripts/UPDATE.sh]: `-DPYTHON_EXECUTABLE=/usr/bin/python3` should do the trick

Comment (paskino) [reply scripts/UPDATE.sh]: yes, I understood that after I posted! Thanks. I'm building a new VM from scratch as we speak.

Comment (paskino) [reply scripts/UPDATE.sh]: still, why is `-DPYTHON_EXECUTABLE` to be preferred to `-DPYVER-3`?
I am using that in the [github action](https://github.com/paskino/SIRF-SuperBuild/blob/build-github-action/.github/workflows/c-cpp.yml#L30) I made.

Comment (KrisThielemans) [reply scripts/UPDATE.sh]: in a controlled environment `PYVER=3` is fine, but once you have multiple python's, `PYTHON_EXECUTABLE` is far more reliable. It is also standard (while `PYVER` is our own work-around, and actually now obsolete)

File: scripts/bootstrap_ubuntu.sh [hunk 3]
```diff
@@ -64,6 +64,7 @@ cd $userHOME/devel
 if [ ! -d $userHOME/devel/SyneRBI_VM ]; then
   git clone https://github.com/SyneRBI/SyneRBI_VM.git
   cd SyneRBI_VM
+  git checkout python3_build
```
Comment (KrisThielemans) [inline scripts/bootstrap_ubuntu.sh:67]: this will need to be removed before merging

File: scripts/bootstrap_ubuntu.sh [hunk 4]
```diff
@@ -79,4 +80,4 @@ bash $userHOME/devel/SyneRBI_VM/scripts/INSTALL_python_packages.sh
 # Therefore no browser is installed on the VM
 
 chown -R $SIRFUSERNAME:users $userHOME
-sudo -u $SIRFUSERNAME -H bash $userHOME/devel/SyneRBI_VM/scripts/UPDATE.sh
+sudo -u $SIRFUSERNAME -H bash $userHOME/devel/SyneRBI_VM/scripts/UPDATE.sh -t update_cil_2011
```
Comment (KrisThielemans) [inline scripts/bootstrap_ubuntu.sh:83]: this will need to be removed before merging. (not sure why it's here)

**Review by KrisThielemans — CHANGES_REQUESTED** [review]
looks good to me. aside from these 2, ready to merge

**Review by KrisThielemans — CHANGES_REQUESTED** [review]
Two comment that remain to be resolved. In addition, https://github.com/SyneRBI/SyneRBI_VM/blob/master/documentation/Getting-started-with-SIRF.md needs updating as it says `python` and talks about `spyder`  versions.

**Pull request merged**